clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -4139,6 +4164,12 @@ scheduler_tmpl_t<RecordType, ReaderType>::eof_or_idle(output_ordinal_t output,
         assert(options_.mapping != MAP_AS_PREVIOUSLY || outputs_[output].at_eof);
         return sched_type_t::STATUS_EOF;
     }
+    if (options_.mapping == MAP_TO_ANY_OUTPUT &&
+        live_inputs <=
+            static_cast<int>(inputs_.size() * options_.exit_if_fraction_left)) {
+        VPRINT(this, 1, "output %d exiting early\n", output);
```

Comment:
Same suggestion as above to add the live_inputs count to the message.

Reply:
Done.